components/dashboards/org.wso2.carbon.dashboards.core/src/main/java/org/wso2/carbon/dashboards/core/DashboardMetadataProvider.java

```diff
@@ -42,13 +42,16 @@
      */
     Optional<DashboardMetadata> get(String dashboardUrl) throws DashboardException;
 
-    Set<DashboardMetadata> getAll() throws DashboardException;
+    Response getDashboardByUser(String user, String dashboardUrl, Optional<String> originComponent) throws
```

Comment:
this is an OSGi component, hence returning a `javax.ws.rs.core.Response` is not appropriate.